pytensor/tensor/slinalg.py

```diff
@@ -67,15 +68,27 @@ def make_node(self, x):
         return Apply(self, [x], [x.type()])
 
     def perform(self, node, inputs, outputs):
-        x = inputs[0]
-        z = outputs[0]
+        (x,) = inputs
+        (z,) = outputs
+
         try:
-            z[0] = scipy.linalg.cholesky(x, lower=self.lower).astype(x.dtype)
+            if x.flags["C_CONTIGUOUS"]:
```

Comment:
I figured it's no harm either way? I'm pretty fuzzy on all the costs/benefits of checking everything is C/Fortran contiguous.

Reply:
The C contiguous case is the one thay needs special care? I would imagine that's the default one most of the times

Reply:
It's the default, but it has to be in fortran order for the overwrite to trigger. Otherwise it gets copied as it's getting shipped out to LAPACK.

It's also possible to end up with non-contiguous arrays via things like concatenation I think? Maybe not on a Pytensor graph though. I'm speaking from experience working with numba + numpy.

Reply:
Why don't they do the transpose trick themselves when it's C contiguous?

Reply:
They have to do it at some point, otherwise the answer will come out wrong. I'm not sure where it happens.